### PR TITLE
Invalidate sessions on password change

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=51523

### CODE REVIEW

File: api/src/org/labkey/api/security/UserManager.java [hunk 1]
```diff
@@ -545,25 +552,72 @@ public void sessionCreated(HttpSessionEvent event)
...
+            {
+                //noinspection DataFlowIssue
+                LOG.debug("Invalidated {} for {}.", StringUtilsLabKey.pluralize(count, "session"), user.getEmail());
+            }
+        });
+    }
+
+    public interface SessionHandler
+    {
+        // Return true to increment the count passed to complete()
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/security/UserManager.java]:
```suggestion
        // Return true to increment the count passed to complete() after handleSession() has been invoked for each active session
```

File: api/src/org/labkey/api/security/UserManager.java [hunk 2]
```diff
@@ -545,25 +552,72 @@ public void sessionCreated(HttpSessionEvent event)
...
+        boolean handleSession(HttpSession session);
+        // Called after all sessions have been passed to handleSession(), only if user is non-null and was logged in
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/security/UserManager.java]:
```suggestion
        // Called after all sessions have been passed to handleSession(), only if user is non-null and was logged in.
        // Useful for coalescing logging, for example.
```

**Review by labkey-jeckels — APPROVED** [review]

**Comment by labkey-jeckels** [comment]
It took me a little while to understand the `SessionHandler` interface so I suggested some additional comments. See if you think they're helpful or just clutter.

**Comment by labkey-adam** [comment]
> It took me a little while to understand the `SessionHandler` interface so I suggested some additional comments. See if you think they're helpful or just clutter.

Yeah, it's not the most obvious interface ever. More docs don't hurt, IMO.

**Pull request merged**